Y-Data Profiling PYTHON Project

Objective
 

In this checkpoint, we will analyze the Diabetes Dataset, which contains medical and demographic information related to diabetes.
 

Dataset description:

 This dataset includes medical measurements and indicators that can help in diagnosing diabetes. The goal is to explore the dataset and uncover any patterns, correlations, and potential issues such as missing values or outliers.
 

Dataset Columns Description

Pregnancies: Number of times the patient was pregnant.
Glucose: Plasma glucose concentration (2 hours in an oral glucose tolerance test).
BloodPressure: Diastolic blood pressure (mm Hg).
SkinThickness: Triceps skinfold thickness (mm).
Insulin: 2-Hour serum insulin (mu U/ml).
BMI: Body Mass Index (weight in kg/(height in m)^2).
DiabetesPedigreeFunction: Diabetes pedigree function (a measure of genetic influence).
Age: Age of the patient (years).
Outcome: Diabetes diagnosis outcome (0 = No, 1 = Yes).

In [4]:
# Load pandas
import pandas as pd

In [6]:
# Use Pandas to load the dataset into a DataFrame.
df = pd.read_csv(r"C:\Users\LENOVO\Downloads\diabetes (1).csv")

In [11]:
#Load Dataset
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [12]:
# Ensure the dataset loads properly by checking the first few rows with head().
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [13]:
# Display the dataset's structure, including column names, data types, and memory usage.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [16]:
# Identify the number of missing values or zeros in the dataset.
df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [18]:
# Summary statistics for each column (mean, min, max, quartiles).
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [19]:
# Look for irregularities, such as columns with unrealistic minimum or maximum values.
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [30]:
# These indices are suceptible to extremes as they are suppose to be in a range (not absolute 0 or higher than a level for a living human)
indicies = ['Glucose', "BloodPressure", 'SkinThickness', 'Insulin', 'BMI', 'DiabetesPedigreeFunction', 'Age']
extremes = df[(df[indicies] == 0).any(axis=1)]
extremes

Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
1        0             80             22       90   0.0   
2      115              0             19      100  27.9   

   DiabetesPedigreeFunction  Age  
1                     0.351   30  
2                     0.000   35

In [ ]:
import ydata_profiling

In [ ]:
ydata_profiling.ProfileReport(df)

In [ ]:
(df['BMI'] == 0).sum()

In [ ]:
(df['BMI'] == 0).sum()

In [ ]:
(df['Glucose'] == 0).sum()

In [ ]:
(df['Insulin'] == 0).sum()

Summary Report and Findings
Dataset Overview
The dataset consists of 768 entries and 9 columns, with 8 numerical variables and 1 categorical variable. This analysis was conducted using pandas for data exploration and ydata profiling for deeper insights.

Key Findings from Pandas Data Exploration:

Data Integrity:
There are no missing values or duplicate rows in the dataset.
However, the dataset contains a significant number of zero values across various numerical columns, which raises concerns about the validity of certain measurements.
Unrealistic Minimum Values:
Glucose: A glucose level of 0 mg/dL is not compatible with life. Normal blood glucose levels typically range from 70 to 100 mg/dL.
Blood Pressure: A diastolic blood pressure level of 0 mmHg is not possible. Normal diastolic values typically range from 60 to 80 mmHg.
Skin Thickness: A skin thickness measurement of 0 mm is unrealistic. Skin thickness can vary based on factors such as age and health conditions.
Insulin: While a person can have an insulin level of 0, particularly in conditions like type 1 diabetes, 48.7% of the dataset has an insulin level of zero, raising concerns about the integrity of this variable. Additionally, an insulin level of 846 is noted, which is not feasible and indicates the presence of outliers.
BMI: Given that the formula for calculating BMI is ( \text{weight in kg} / (\text{height in m})^2 ), a BMI of 0 is not feasible, as weight cannot be zero in a living person.
Insights from Ydata Profiling
Pregnancies: The analysis shows a downward trend in the number of pregnancies, indicating that more individuals in the dataset have fewer pregnancies.

Glucose Levels: The distribution of glucose levels exhibits a hill-shaped trend, with a higher concentration of individuals having glucose levels in the middle range of 75-200 mg/dL.

Insulin Levels: Similar to glucose, insulin levels also show a hill-shaped trend, with a significant number of individuals falling within the middle range.

Correlation Analysis:
Age vs. Insulin: The correlation chart indicates that most individuals in their 20s and 30s have an insulin level around 150, while those in their 50s and 60s have higher average insulin levels, around 300-350.
Insulin vs. Outcome: The correlation heatmap shows a light correlation between insulin levels and the outcome variable.
Strong Correlations:
Age and Pregnancies: A strong correlation exists, suggesting that as age increases, the number of pregnancies tends to decrease.
Insulin and Skin Thickness: A strong correlation indicates that higher insulin levels may be associated with increased skin thickness.
Glucose and Outcome: A strong correlation suggests that glucose levels may significantly impact the outcome variable.
BMI and Skin Thickness: A strong correlation indicates that higher BMI is associated with increased skin thickness.
Conclusion
The presence of unrealistic minimum values, particularly for glucose, blood pressure, skin thickness, insulin, and BMI, suggests cleaning and validation. The trends observed in pregnancies, glucose, and insulin levels provide valuable information for understanding the population's health status. Furthermore, the correlation analysis highlights significant relationships that could inform future research and health interventions.

Recommendations
Data Cleaning: Address the unrealistic minimum values of BloodPressure, Glucose and SkinThickness and the high percentage of zero values in the insulin variable. I suggest we replace the zeros of the insulin variable with the mode of the variable reason being that, there are presense of outliers in the variable as stated above.